### ISSUE TITLE
ch-grow lark error: wrong "lark" dependency listed in documentation.

### ISSUE THREAD

**Issue by j-ogas** [hpc/charliecloud#494]
I appear to have broken ch-grow in my test box environment. It no longer works. Has this issue been seen before?

```
$ pip3 list
[...]
lark                     0.0.4
lark-parser              0.7.1
[...]
$ ./bin/ch-run --version
0.10
$ ./bin/ch-build --builder ch-grow -t debian9 -f ./test/Dockerfile.debian9 ./test
building with: ch-grow
Traceback (most recent call last):
  File "/home/jogas/upstream/bin/ch-grow", line 177, in <module>
    class Main_Loop(lark.Visitor):
AttributeError: module 'lark' has no attribute 'Visitor'
```

**Comment by j-ogas** [comment]
The issue was that I installed the _wrong lark_.

The incorrect way to install lark (what I did):

```
$ pip3 install lark-parser --user
```

The correct way:
```
$ pip3 install lark_parser --user
```
```

**Comment by j-ogas** [comment]
Turns out I'm not crazy (I hope). We appear to list `lark-parser` as the correct dependency in our install documentation.

**Comment by reidpr** [comment]
I'm not sure what's going on here. `lark-parser` (with hyphen) works for me:

```
$ pip3 uninstall lark-parser
[...]
  Successfully uninstalled lark-parser-0.7.1
$ bin/ch-grow --storage=$CH_TEST_TARDIR/_ch-grow -t foo -f test/Dockerfile.alpine39 test
missing dependency: Python module "lark"
$ pip3 install lark-parser
Collecting lark-parser
Installing collected packages: lark-parser
Successfully installed lark-parser-0.7.1
$ bin/ch-grow --storage=$CH_TEST_TARDIR/_ch-grow -t foo -f test/Dockerfile.alpine39 test
growing: /scratch/tarballs/_ch-grow/img/foo:latest
  2 FROM alpine:3.9
  4 RUN ['/bin/sh', '-c', 'apk add --no-cache bc']
fetch http://dl-cdn.alpinelinux.org/alpine/v3.9/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.9/community/x86_64/APKINDEX.tar.gz
(1/5) Installing ncurses-terminfo-base (6.1_p20190105-r0)
(2/5) Installing ncurses-terminfo (6.1_p20190105-r0)
(3/5) Installing ncurses-libs (6.1_p20190105-r0)
(4/5) Installing readline (7.0.003-r1)
(5/5) Installing bc (1.07.1-r0)
Executing busybox-1.29.3-r10.trigger
OK: 13 MiB in 19 packages
  7 CMD (unimplemented)
```

**Comment by reidpr** [comment]
You do have `lark` listed there which is definitely wrong.

**Comment by reidpr** [comment]
Does the underscore vs. hyphen matter if you uninstall `lark`?

**Comment by j-ogas** [comment]
I am not able to reproduce the error in the original context, e.g., both `lark` and `lark-parser` installed. I can however reproduce the error text by removing the lark parser module and leaving only `lark`.

I'm convinced this is issue is erroneous.

**Comment by reidpr** [comment]
Both `lark` and `lark-parser` offer a module named `lark`. I wonder if it is non-deterministic which one you get if you have both installed, and installing multiple times to test underscore vs. hyphen caused different results.

It could be nice to give a better error message if we get the wrong `lark` (which is for "transforming a HTTP request into a redis command").

**Comment by rstyd** [comment]
Perhaps we could use `inspect` or something similar to identift an attribute found in `lark` but not `lark-parser`. If the ` import lark` module contains that attribute, it's the wrong version.

**Comment by reidpr** [comment]
The bad `lark` doesn't have much of anything at the module (package?) level. We could just use `lark.Visitor`.

Bad:

```
>>> dir(lark)
['__builtins__', '__cached__', '__doc__', '__file__', '__loader__', '__name__', '__package__', '__path__', '__spec__']
```

Good:

```
>>> dir(lark)
['Discard', 'GrammarError', 'InlineTransformer', 'Lark', 'LexError', 'ParseError', 'Token', 'Transformer', 'Tree', 'UnexpectedCharacters', 'UnexpectedInput', 'UnexpectedToken', 'Visitor', '__builtins__', '__cached__', '__doc__', '__file__', '__loader__', '__name__', '__package__', '__path__', '__spec__', '__version__', 'common', 'exceptions', 'grammar', 'inline_args', 'lark', 'lexer', 'load_grammar', 'parse_tree_builder', 'parser_frontends', 'parsers', 'tree', 'utils', 'v_args', 'visitors']
```

**Comment by reidpr** [comment]
I believe the underlying confusion is fixed in PR #496. Closing.
